src/main/java/optic_fusion1/mcantimalware/DirectoryWatcherService.java

```diff
@@ -74,6 +74,7 @@ public DirectoryWatcherService(Main main, Path dir, boolean recursive) throws IO
     private synchronized void addFileToProcess(File file) {
         if (file.isDirectory()) {
             File[] files = file.listFiles();
+            assert files != null;

```

Comment:
Assertions are removed at runtime unless you explicitly specify to "enable assertions" when compiling your code. Java Assertions are not to be used on production code and should be restricted to private methods.

At least that's what i found, not sure if that's changed for recent java versions, probably not

Reply:
so it likes the original? Umm I saw my IDE warn and choose that as a fix, at least it did :))

Reply:
My IDE doesn't warn be about that?
If anything it should return, not use something that shouldn't be used in production code.
Various threads say it shouldn't be used in production code, only testing
https://stackoverflow.com/a/2758645
https://stackoverflow.com/a/18907607
https://softwareengineering.stackexchange.com/a/137163
https://softwareengineering.stackexchange.com/a/137167
Assert is pretty much useless as well unless you enable assertions with the -ea flag, and i'd rather not have that flag be a requirement

Reply:
Oh sorry, I did not know about that. I have reverted that commit